web/book/tests/snapshots/snapshot__tests__prql__examples__variables-1.prql.snap

```diff
@@ -1,24 +1,30 @@
 ---
-source: book/tests/snapshot.rs
+source: web/book/tests/snapshot.rs
 expression: "from employees\ngroup [emp_no] (\n  aggregate [\n    emp_salary = average salary     # average salary resolves to \"AVG(salary)\" (from stdlib)\n  ]\n)\njoin titles [==emp_no]\ngroup [title] (\n  aggregate [\n    avg_salary = average emp_salary\n  ]\n)\nselect salary_k = avg_salary / 1000 # avg_salary should resolve to \"AVG(emp_salary)\"\ntake 10                             # induces new SELECT\nderive salary = salary_k * 1000     # salary_k should not resolve to \"avg_salary / 1000\"\n"
 ---
-WITH table_1 AS (
+WITH table_3 AS (
   SELECT
-    AVG(salary) AS _expr_0,
+    AVG(salary) AS _expr_1,
     emp_no
   FROM
     employees
   GROUP BY
     emp_no
+),
+table_1 AS (
+  SELECT
+    AVG(table_2._expr_1) AS _expr_0
+  FROM
+    table_3 AS table_2
+    JOIN titles ON table_2.emp_no = titles.emp_no
+  GROUP BY
+    titles.title
 )
 SELECT
-  AVG(table_0._expr_0) / 1000 AS salary_k,
-  AVG(table_0._expr_0) / 1000 * 1000 AS salary
+  _expr_0 / 1000 AS salary_k,
+  _expr_0 / 1000 * 1000 AS salary
 FROM
   table_1 AS table_0
-  JOIN titles ON table_0.emp_no = titles.emp_no
-GROUP BY
-  titles.title
 LIMIT
   10
```

Comment:
This is a good example of the changes — the SQL is a bit less succinct:
- We have an extra CTE
- `FROM table_3 AS table_2`

...but not in a dramatic way. It's possible an inevitable part of making the compiler pipeline more modularized — the whole thing becomes a bit less optimized, and it's probably a good tradeoff

Reply:
I don't think that these things will cause performance degradation to any modern optimizer. But it is admittedly less readable.

(My tolerance for badly looking target SQL went way way up since I started working for edgedb :D )

Reply:
`FROM table_3 AS table_2` could be fixed probably - it's making a distinction between a "table name" and a "table instance name" (in Postgres they call these two "relation" and "range var"). We could use "table name" as the "table instance name" here. But not in general, because same table, could be instanced in a query twice:

```
WITH tab AS (SELECT ...)
SELECT a.id, b.id
FROM tab a
JOIN tab b ON (a.id = b.parent_id)
```

Reply:
The extra CTE is also not required here, but it is required in a very similar case. I didn't think it was worth the time to find a rule that would separate these cases correctly.

In this PR, I've basically declared "computations of newly-aggregated expressions are not allowed". I needed this rule to make another query work, but it was a bit to strict because is needlessly affecting this query.